app/question.js

```diff
@@ -58,30 +73,73 @@ define(['jquery', 'app/messages', 'app/config', 'app/string'], function ($, mess
   }
 
   // Handle how to move to the next question
-  // depending on the tutorial status.
+  // depending on the tutorial status and algorithm.
   function nextQuestion() {
-    if (!inTutorial) {
-      items.splice(currentItemIndex,1);
-      currentItemIndex %= items.length;
-    } else {
-      currentItemIndex++;
-      inTutorial = checkTutorialStatus();
+    switch(config.constant("ALGORITHM")) {
+      case "flashcard":
+        // Use flaschcard method to determine next question
```

Comment:
In the bodies of the cases you should only call the correct method, selecting what to do is a different level of abstraction from actually doing it, methods should operate at one level of abstraction.